sqlgen/db.go

```diff
@@ -92,7 +94,65 @@ func NewDB(conn *sql.DB, schema *Schema) *DB {
 	return db
 }
 
+// WithShardLimit scopes the DB to only allow queries with the given key-value
+// pairs. This means any query must include a filter for the key-value pairs in
+// the limit, and any write must have columns including the specified key-value
+// pairs.
+func (db *DB) WithShardLimit(shardLimit Filter) (*DB, error) {
+	if db.shardLimit != nil {
+		return nil, errors.New("already limited")
+	}
+
+	dbCopy := *db
+	dbCopy.shardLimit = shardLimit
+	return &dbCopy, nil
+}
+
+func (db *DB) checkFilterAgainstShardLimit(filter Filter) error {
+	if db.shardLimit == nil {
+		return nil
+	}
+	for k, v := range db.shardLimit {
+		filterV, ok := filter[k]
+		if !ok {
+			return fmt.Errorf("db is sharded to require %s = %v, but query does not filter on %s", k, v, k)
+		}
+		if filterV != v {
```

Comment:
oh... int64(orgid) != int(orgid). this might break things

Reply:
i think unit tests should catch all breakage